packages/paste-core/components/form-pill-group/src/PillCloseIcon.tsx

```diff
@@ -1,20 +1,55 @@
 import * as React from 'react';
 import {Box} from '@twilio-paste/box';
+import type {BoxProps} from '@twilio-paste/box';
 import {CloseIcon} from '@twilio-paste/icons/esm/CloseIcon';
 
 interface PillCloseIconProps {
-  onClick: () => void;
+  onClick?: () => void;
+  selected?: boolean;
+  variant?: 'error' | 'default';
+  element?: BoxProps['element'];
 }
 
-export const PillCloseIcon: React.FC<PillCloseIconProps> = ({onClick}) => {
+export const PillCloseIcon: React.FC<PillCloseIconProps> = ({
+  element = 'FORM_PILL_CLOSE',
+  onClick = () => {},
+  selected = false,
+  variant = 'default',
+}) => {
   return (
     <Box
+      element={element}
       onClick={onClick}
+      position="absolute"
+      right={0}
+      top={0}
+      display="flex"
+      alignItems="center"
+      justifyContent="center"
+      height="24px"
+      width="24px"
+      borderRadius="borderRadiusCircle"
+      borderWidth="borderWidth10"
+      borderStyle="solid"
+      borderColor="transparent"
+      transition="color 150ms ease-in, border-color 150ms ease-in"
+      color="colorTextIcon"
       _hover={{
         cursor: 'pointer',
+        borderColor: selected
+          ? 'transparent'
+          : variant === 'error'
+          ? 'colorBorderErrorStronger'
+          : 'colorBorderPrimaryStronger',
+        backgroundColor: selected
+          ? variant === 'error'
```

Comment:
Note: I normally don't like double ternaries. In fact I think there's an ESLint rule to prevent these. However I was iterating rapidly and didn't want to keep shuffling if/else logic around. This could use a second pair of eyes once designs are solidified.